qubes/api/admin.py

```diff
@@ -486,6 +487,32 @@ def vm_volume_resize(self, untrusted_payload):
         finally:  # even if calling qubes.ResizeDisk inside the VM failed
             self.app.save()
 
+    @qubes.api.method('admin.vm.volume.Clear', no_payload=True,
+        scope='local', write=True)
+    @asyncio.coroutine
+    def vm_volume_clear(self):
+        self.enforce(self.arg in self.dest.volumes.keys())
+
+        self.fire_event_for_permission()
+
+        volume = self.dest.volumes[self.arg]
+        size = volume.size
+
+        # Clear the volume by importing empty data into it
+        path = yield from volume.import_data(size)
```

Comment:
Technically, `volume.import_data` doesn't need to be a coroutine, use `qubes.utils.coro_maybe()` helper, or `self.dest.storage.import_data()` wrapper.
Same applies to import_data_end.